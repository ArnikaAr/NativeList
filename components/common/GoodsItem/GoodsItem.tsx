import React, { FC, useState } from 'react';
import { StyleSheet, View } from 'react-native';
import DeleteModal from '../../modals/DeleteModal/DeleteModal';
import { Chip } from 'react-native-paper';
import store from '../../../src/store/index'
import { Item } from '../../../src/models';
import { observer } from 'mobx-react-lite';
export interface IGoodsItem {
    item: Item, 
}

const GoodsItem: FC<IGoodsItem> = ({
    item, 
}) => {

    const [visible, setVisible] = useState(false);

    const showModal = () => setVisible(true);
    const hideModal = () => setVisible(false);

    const removeItem = (item: Item): void => {
        store.removeItem(item)
    }
    return (
        <View key={item.id} style={styles.chipStyles}>
            <Chip closeIcon="close" onClose={() => showModal()}>{item.name}</Chip>
            <DeleteModal visible={visible} hideModal={hideModal} modalHandler={() => removeItem(item)} />
        </View>
    );
};

export default observer(GoodsItem);
const styles = StyleSheet.create({
    chipStyles: {
        margin: 5
    },
});