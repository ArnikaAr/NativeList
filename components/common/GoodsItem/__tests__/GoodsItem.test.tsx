import React from 'react';
import renderer from 'react-test-renderer';
import GoodsItem from '../GoodsItem';
import { render } from '@testing-library/react-native';

const testItem = {
    key: '1', 
    value: 'test value'
};

describe('<ItemBlock />', () => {
    it('ItemBlock match snapshot', () => {
        const tree = renderer.create(<GoodsItem item={testItem} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('ItemBlock contain  right info', () => {
        const {getByText } = render(
            <GoodsItem item={testItem} />,
          );
          const listName = getByText("test value");
          expect(listName).toBeTruthy();
    });
});

