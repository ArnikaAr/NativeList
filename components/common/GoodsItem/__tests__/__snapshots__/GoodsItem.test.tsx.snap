// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemBlock /> ItemBlock match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#6b4faa",
      "borderRadius": 7,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 35,
      "justifyContent": "flex-start",
      "margin": 5,
      "padding": 5,
    }
  }
>
  <Text>
    test value
  </Text>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 15.5,
          "height": 20,
          "margin": 0,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": 20,
        }
      }
      testID="icon-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 15.5,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "padding": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 15.5,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "#6b4faa",
                  "fontSize": 15,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 15.5,
          "height": 20,
          "margin": 0,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": 20,
        }
      }
      testID="icon-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 15.5,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "padding": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 15.5,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "#6b4faa",
                  "fontSize": 15,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onDismiss={[Function]}
    transparent={true}
    visible={false}
  />
  <Modal
    hardwareAccelerated={false}
    onDismiss={[Function]}
    transparent={true}
    visible={false}
  />
</View>
`;
